### DISCUSSION TITLE
Strange behaviour: Procedural filter not executing on initial try

### DISCUSSION THREAD

**Discussion by u-RraaLL** [uBlockOrigin/uBlock-issues#1884] [Q&A] [answered]
I've run into this strange behavior earlier, when I was trying to solve [this](https://www.reddit.com/r/uBlockOrigin/comments/rikoep/overlaypopups_make_reddit_unusable_without_login/).

<details><summary>Here's a screen recording:</summary>

https://user-images.githubusercontent.com/92610420/146639883-e92b5f8f-66d7-4678-b616-a654e2378cdd.mp4
</details>

Any ideas why this would happen?

I've tried on my normal profile (latest dev uBO) as well as on a new one, with latest stable uBO. Happens on both.

<details><summary>New profile config</summary>

```yml
uBlock Origin: 1.39.2
Firefox: 95
filterset (summary): 
  network: 77511
  cosmetic: 40063
  scriptlet: 16339
  html: 608
listset (total-discarded, last updated): 
  default: 
    ublock-filters: 30756-31, 2h.11m
    ublock-badware: 3573-1, 2h.11m
    ublock-privacy: 184-0, 2h.11m
    ublock-abuse: 81-0, 2h.11m
    ublock-unbreak: 1738-0, 2h.11m
    easylist: 62045-559, 2h.11m
    easyprivacy: 26508-475, 2h.11m
    urlhaus-1: 7162-0, 2h.11m
    plowe-0: 3691-3, 2h.11m
filterset (user): [empty]
modifiedUserSettings: [none]
modifiedHiddenSettings: [none]
supportStats: 
  launchToReadiness: 441
  launchFromSelfie: false
```
</details>

**Comment by gorhill** [comment]
Maybe the `src` attribute is changed without any other DOM layout changes? For these specific cases, there is the [`watch-attr` operator](https://github.com/gorhill/uBlock/wiki/Procedural-cosmetic-filters#subjectwatch-attrarg) -- I will show you how to use it if you can paste here the actual filter which you say is problematic (transcribing from video is tedious).

**Reply by u-RraaLL** [reply]
Yes, sorry about that:

    reddit.com##iframe[src$="actionSource=header_signup"]:upward(2)

**Reply by gorhill** [reply]
Try:

    reddit.com##iframe:watch-attr(src):iframe[src$="actionSource=header_signup"]:upward(2)

**Reply by u-RraaLL** [reply]
It's highlighted as invalid in My filters and filter lists report 0 of 0 used.

**Reply by gorhill** [reply]
Yes sorry I didn't try it as I can't reproduce the issue on my side. What do I need to do to reproduce this?

**Reply by u-RraaLL** [reply]
1. Add the filter to My filters.
2. Open https://www.reddit.com/r/uBlockOrigin/new/ in a private window or a new browser profile to be logged out.
3. Wait a few seconds for reddit to finish loading. <-if clicked earlier, the filter will usually work
4. Click [Register] in the top bar.
5. See the popup.

**Comment by gorhill** [comment, marked as answer]
Maybe:

    reddit.com##iframe:watch-attr(src):upward(1) > iframe[src$="actionSource=header_signup"]:upward(2)

Or better if that works, requires browser support for `:is()`:

    reddit.com##iframe:watch-attr(src):is([src$="actionSource=header_signup"]):upward(2)

**Reply by u-RraaLL** [reply]
Seems to be working. Thanks.

**Reply by gorhill** [reply]
I added another suggestion, semantically better.

**Reply by u-RraaLL** [reply]
Ok, thanks.

**Reply by u-RraaLL** [reply]
> requires browser support for `:is()`

Don't modern browsers all support it nowadays? (except for Safari) 

I remember even seeing it used in this commit https://github.com/gorhill/uBlock/commit/eccf613edfe480d34cb225dac203d3213f3ef2f7.

**Reply by uBlock-user** [reply]
https://developer.mozilla.org/en-US/docs/Web/CSS/:is

**Reply by gorhill** [reply]
> Don't modern browsers all support it nowadays?

Yes, while it's the better solution by default, the uglier solution will still need to be suggested for users who report still using older browser version, FF78 for example. The use in uBO's CSS will just cause minor visual glitch, as opposed to a completely not-working procedural filter.
